pkg/operator/ceph/nfs/nfs.go

```diff
@@ -305,7 +306,17 @@ func (r *ReconcileCephNFS) configureNFSPool(n *cephv1.CephNFS) error {
 
 	_, err := cephclient.GetPoolDetails(r.context, r.clusterInfo, poolName)
 	if err != nil {
-		return errors.Wrapf(err, "pool %q could not be retrieved, ensure the pool is defined with a CephBlockPool", poolName)
+		// create default `.nfs` pool for versions >= ceph v16.2.7, if it does not exist.
+		if r.clusterInfo.CephVersion.IsAtLeast(cephNFSChangeVersion) && strings.Contains(err.Error(), "ENOENT") {
```

Comment:
Let's use `if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {` instead of checking the error string

Reply:
> Let's use `if code, ok := exec.ExitStatus(err); ok && code == int(syscall.ENOENT) {` instead of checking the error string

The function is not able to extract ENOENT
my code looks like
```golang
	_, err := cephclient.GetPoolDetails(r.context, r.clusterInfo, poolName)
	if err != nil {
		code, ok := exec.ExitStatus(err)
		fmt.Println(code, ok, err)
		// create default `.nfs` pool if it does not exist for ceph versions >= v16.2.7.
		if (ok && code == int(syscall.ENOENT)) && r.clusterInfo.CephVersion.IsAtLeast(cephNFSChangeVersion) {
			args := []string{"osd", "pool", "create", poolName}
			output, err := cephclient.NewCephCommand(r.context, r.clusterInfo, args).Run()
			if err != nil {
				return errors.Wrapf(err, "failed to create default nfs pool %q. %s", poolName, string(output))
			}
		} else {
			return errors.Wrapf(err, "pool %q could not be retrieved, ensure the pool is defined with a CephBlockPool",
				poolName)
		}
	}
```

```
2022-04-12 11:04:33.848414 D | exec: Running command: ceph osd pool get .nfs all --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json
0 false failed to get pool .nfs details. . Error ENOENT: unrecognized pool '.nfs': exit status 2
2022-04-12 11:04:34.239527 E | ceph-nfs-controller: failed to reconcile CephNFS "rook-ceph/my-nfs". failed to configure nfs pool ".nfs": pool ".nfs" could not be retrieved, ensure the pool is defined with a CephBlockPool: failed to get pool .nfs details. . Error ENOENT: unrecognized pool '.nfs': exit status 2
```

Reply:
Maybe because the error returned from `GetPoolDetails` is wrapped? Try to just have `GetPoolDetails()` return `err`.

Reply:
We can also unwrap the error and check it?

Reply:
negative, neither work.
`Error ENOENT` comes from the output string and `exit status 2` is the unwrapped error which does not seem to be detected as `ENOENT`